tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -545,4 +724,214 @@ TEST_F(ChargepointTestFixtureV201, K01FR49_IfNumberPhasesMissingForACEVSE_ThenSe
     EXPECT_THAT(numberPhases, testing::Eq(3));
 }
 
+TEST_F(ChargepointTestFixtureV201, K01FR49_IfNumberPhasesMissingForACChargingStation_ThenSetNumberPhasesToThree) {
+    device_model->set_value(ControllerComponentVariables::ChargingStationSupplyPhases.component,
+                            ControllerComponentVariables::ChargingStationSupplyPhases.variable.value(),
+                            AttributeEnum::Actual, std::to_string(3), true);
+
+    auto periods = create_charging_schedule_periods(0);
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), DEFAULT_TX_ID);
+
+    auto sut = handler.validate_profile_schedules(profile, std::nullopt);
+
+    auto numberPhases = profile.chargingSchedule[0].chargingSchedulePeriod[0].numberPhases;
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::Valid));
+    EXPECT_THAT(numberPhases, testing::Eq(3));
+}
+
+TEST_F(ChargepointTestFixtureV201, K01FR06_ExisitingProfileLastForever_RejectIncoming) {
+    install_profile_on_evse(DEFAULT_EVSE_ID, DEFAULT_PROFILE_ID, ocpp::DateTime(date::utc_clock::time_point::min()),
+                            ocpp::DateTime(date::utc_clock::time_point::max()));
+
+    auto periods = create_charging_schedule_periods(0);
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID + 1, ChargingProfilePurposeEnum::TxDefaultProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), {},
+        ChargingProfileKindEnum::Absolute, DEFAULT_STACK_LEVEL, ocpp::DateTime("2024-01-02T13:00:00"),
+        ocpp::DateTime("2024-03-01T13:00:00"));
```

Comment:
I'm curious as to why we use `utc_clock::time_point::max/min` when we want to say "undefined" in `install_profile_on_evse` but `{}` in `create_charging_profile`. It seems like it would be much easier to keep it consistent.

Reply:
`install_profile_on_evse()` bypasses our validation/conforming functions, so we need to store the profile as if it was already modified for storage and later checks. In contrast, `create_charging_profile()` represents actual data that may be passed into `validate_profile()`. Thus, the values are missing instead of being set to `min()` or `max()`.

Reply:
Please add this as a comment to the source code as well